### ISSUE TITLE
[bug] Cannot generate examples for /__.__/

### ISSUE THREAD

**Issue by taichi-ishitani** [tom-lord/regexp-examples#30]
Hi,

I found an issue that `#examples` method cannot generate examples for regexp `/__.__/`.

```
irb(main):017:0> /__.__/.examples
=> []
```

Note:

Examples for following regexp can be generated.

* `/____/`
* `/_._/`
* `/_.__/`
* `/__._/`

```
irb(main):018:0> /____/.examples
=> ["____"]
irb(main):019:0> /_._/.examples
=> ["_a_", "_b_", "_c_", "_d_", "_e_"]
irb(main):020:0> /_.__/.examples
=> ["_a__", "_b__", "_c__", "_d__", "_e__"]
irb(main):021:0> /__._/.examples
=> ["__a_", "__b_", "__c_", "__d_", "__e_"]
```

My environment:

* ruby 2.5.3p105 (2018-10-18 revision 65156) [x64-mingw32]
* regexp-examples (1.4.4)

**Comment by tom-lord** [comment]
I wonder, did you encounter this in a real world scenario or because you were intentionally looking for a bug? 😉 

This is caused by the "hacky" implementation of generating matches for regexes with back-references.  Due to [this code](https://github.com/tom-lord/regexp-examples/blob/9d7aff092a604b38b0a7937804159c4115fd6639/lib/regexp-examples/groups.rb#L180-L187), I effectively say "any instance of `__xx__` in the string is a placeholder, not a real part of the result.

Possible quick "fix": Make the placeholder longer/more obscure. It could be 100 "random" characters long, for all I care!

**Comment by taichi-ishitani** [comment]
Hi @tom-lord ,
Thanks for your comment.

> I wonder, did you encounter thi

I'm using your gem as random string generator for RSpec examples.
When I tried to generate a hidden method name like `__a__` by using `/__\w__/` regexp then I got this problem.

I have workaround for this bug; generate random string by using `/\w/` regexp and then add `__` to head and tail of generated string. Therefore I have no problem due to this bug.
https://github.com/rggen/rggen-basic-output-components/blob/aba7e720e89146c4223438ecfeb38a46ebd076b1/spec/rggen/basic_output_components/systemverilog_utility/identifier_spec.rb#L126

**Comment by tom-lord** [comment]
Interesting use case, thanks for the report.

As I said above, I think the laziest solution will be to simply make the placeholder pattern much more obscure - so it's virtually impossible to re-encounter this bug unless you intentionally go looking for it 😅 

I'll fix this in a new release soon, when I get time. Should be very straightforward.

**Comment by taichi-ishitani** [comment]
As I said, I have a workaround for this bug. Therefore this is not a priority issue and you don't need to hurry.

**Comment by tom-lord** [comment]
"Fixed" in version `1.5.0` 😃

**Comment by taichi-ishitani** [comment]
Thank you!
